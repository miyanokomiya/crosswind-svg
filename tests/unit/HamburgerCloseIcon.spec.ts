import { mount } from "@vue/test-utils";
import Target from "../../src/components/HamburgerCloseIcon.vue";

describe("HamburgerCloseIcon", () => {
  it("expanded:false", () => {
    const wrapper = mount(Target, {
      propsData: { expanded: false }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("expanded:true", () => {
    const wrapper = mount(Target, {
      propsData: { expanded: true }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
