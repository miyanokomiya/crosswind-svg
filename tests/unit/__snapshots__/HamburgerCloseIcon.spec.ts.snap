// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerCloseIcon expanded:false 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor; user-select: none;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
    stroke-linecap="round"
    stroke-width="4"
  >
    <path
      d="M 20 20 44 20"
      style="transform-origin: 50% 50%; transition: all 0.5s;"
    />
    <path
      d="M 20 32 44 32"
      style="transform-origin: 50% 50%; transition: all 0.5s;"
    />
    <path
      d="M 20 44 44 44"
      style="transform-origin: 50% 50%; transition: all 0.5s;"
    />
  </g>
</svg>
`;

exports[`HamburgerCloseIcon expanded:true 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor; user-select: none;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
    stroke-linecap="round"
    stroke-width="4"
  >
    <path
      d="M 20 20 44 20"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: translateY(12px);"
    />
    <path
      d="M 20 32 44 32"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: scaleX(0);"
    />
    <path
      d="M 20 44 44 44"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: translateY(-12px);"
    />
  </g>
</svg>
`;
